include/revng/ADT/GenericGraph.h

```diff
@@ -996,6 +996,124 @@ class MutableEdgeNode
     return llvm::make_range(predecessors_begin(), predecessors_end());
   }
 
+private:
+  /// Custom child iterator which iterates over the concatenation of the
+  /// successors and predecessors of a node. This is used for the `Undirected`
+  /// `GraphTraits` specialization. This is done taking inspiration from the
+  /// internals of `llvm::concat` implementation, but overcoming the limitation
+  /// causing by the return `EdgeView` going of out scope.
+  template<typename SuccessorIterator, typename PredecessorIterator>
+  class UndirectedChildIterator
+    : public llvm::iterator_facade_base<
+        UndirectedChildIterator<SuccessorIterator, PredecessorIterator>,
+        std::bidirectional_iterator_tag,
+        EdgeView> {
+  private:
+    SuccessorIterator SBegin, SEnd;
+    PredecessorIterator PBegin, PEnd;
```

Comment:
Split in 4 fields

Reply:
Done in https://github.com/revng/revng/blob/f14ffdc32258b98198148f45162302e6cbecc7c1/include/revng/Support/Undirected.h#L51